presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetTypeUtils.java

```diff
@@ -44,27 +64,68 @@ private ParquetTypeUtils()
     public static Optional<RichColumnDescriptor> getDescriptor(MessageType fileSchema, MessageType requestedSchema, List<String> path)
     {
         checkArgument(path.size() >= 1, "Parquet nested path should have at least one component");
+        int index = getPathIndex(fileSchema, requestedSchema, path);
+        return getDescriptor(fileSchema, requestedSchema, index);
+    }
+
+    public static Optional<RichColumnDescriptor> getDescriptor(MessageType fileSchema, MessageType requestedSchema, int index)
+    {
+        if (index == -1) {
+            return empty();
+        }
+        PrimitiveColumnIO columnIO = getColumns(fileSchema, requestedSchema).get(index);
+        ColumnDescriptor descriptor = columnIO.getColumnDescriptor();
+        return Optional.of(new RichColumnDescriptor(descriptor.getPath(), columnIO.getType().asPrimitiveType(), descriptor.getMaxRepetitionLevel(), descriptor.getMaxDefinitionLevel()));
+    }
+
+    private static int getPathIndex(MessageType fileSchema, MessageType requestedSchema, List<String> path)
+    {
         int level = path.size();
-        for (PrimitiveColumnIO columnIO : getColumns(fileSchema, requestedSchema)) {
-            ColumnIO[] fields = columnIO.getPath();
+        List<PrimitiveColumnIO> columns = getColumns(fileSchema, requestedSchema);
+        int index = -1;
+        for (int i = 0; i < columns.size(); i++) {
```

Comment:
Can we rename `i` and `j` to something more descriptive? Maybe `columnIndex` and `level` (and rename `level` to `maxLevel`)?